Tests failing on GHC 9.4 on current Hackage release (0.4.2)

#462 adapted some code to GHC 9.4, specifically some tests, but these changes haven't been released on Hackage. As such, the tests fail on GHC 9.4 on the latest release.

@raehik what error are you seeing?

I wonder if I am running into the same thing, coming via `nix build` w/ GHC 9.4.3:

```
Building library for optics-0.4.2..
[1 of 1] Compiling Optics           ( src/Optics.hs, dist/build/Optics.o, dist/build/Optics.dyn_o )
[1 of 1] Compiling Optics           ( src/Optics.hs, dist/build/Optics.p_o )
Preprocessing test suite 'optics-tests' for optics-0.4.2..
Building test suite 'optics-tests' for optics-0.4.2..
[1 of 9] Compiling Optics.Tests.Properties ( tests/Optics/Tests/Properties.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Properties.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Properties.dyn_o )
[2 of 9] Compiling Optics.Tests.Utils ( tests/Optics/Tests/Utils.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Utils.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Utils.dyn_o )
[3 of 9] Compiling Optics.Tests.Misc ( tests/Optics/Tests/Misc.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Misc.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Misc.dyn_o )
[4 of 9] Compiling Optics.Tests.Labels.TH ( tests/Optics/Tests/Labels/TH.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Labels/TH.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Labels/TH.dyn_o )
[5 of 9] Compiling Optics.Tests.Labels.Generic ( tests/Optics/Tests/Labels/Generic.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Labels/Generic.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Labels/Generic.dyn_o )

tests/Optics/Tests/Labels/Generic.hs:72:15: error:
    Ambiguous occurrence ‘name’
    It could refer to
       either the field ‘name’ of record ‘Human’,
              defined at tests/Optics/Tests/Labels/Generic.hs:27:5
           or the field ‘name’ of record ‘Fish’,
              defined at tests/Optics/Tests/Labels/Generic.hs:22:16
           or the field ‘name’ of record ‘Mammal’,
              defined at tests/Optics/Tests/Labels/Generic.hs:17:11
   |
72 | label1rhs s = name (s :: Human a)
   |               ^^^^

tests/Optics/Tests/Labels/Generic.hs:80:15: error:
    Ambiguous occurrence ‘name’
    It could refer to
       either the field ‘name’ of record ‘Human’,
              defined at tests/Optics/Tests/Labels/Generic.hs:27:5
           or the field ‘name’ of record ‘Fish’,
              defined at tests/Optics/Tests/Labels/Generic.hs:22:16
           or the field ‘name’ of record ‘Mammal’,
              defined at tests/Optics/Tests/Labels/Generic.hs:17:11
   |
80 | label3rhs s = name (fish s :: Fish)
   |               ^^^^
[6 of 9] Compiling Optics.Tests.Eta ( tests/Optics/Tests/Eta.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Eta.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Eta.dyn_o )
[7 of 9] Compiling Optics.Tests.Core ( tests/Optics/Tests/Core.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Core.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Core.dyn_o )
[8 of 9] Compiling Optics.Tests.Computation ( tests/Optics/Tests/Computation.hs, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Computation.o, dist/build/optics-tests/optics-tests-tmp/Optics/Tests/Computation.dyn_o )
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -8,9 +8,9 @@
 #
 # For more information, see https://github.com/haskell-CI/haskell-ci
 #
-# version: 0.16.1
+# version: 0.16.3
 #
-# REGENDATA ("0.16.1",["github","--config=cabal.haskell-ci","cabal.project"])
+# REGENDATA ("0.16.3",["github","--config=cabal.haskell-ci","cabal.project"])
 #
 name: Haskell-CI
 on:
@@ -37,19 +37,19 @@ jobs:
             compilerVersion: "8.4"
             setup-method: hvr-ppa
             allow-failure: false
-          - compiler: ghc-9.6.1
+          - compiler: ghc-9.6.2
             compilerKind: ghc
-            compilerVersion: 9.6.1
+            compilerVersion: 9.6.2
             setup-method: ghcup
             allow-failure: false
-          - compiler: ghc-9.4.4
+          - compiler: ghc-9.4.5
             compilerKind: ghc
-            compilerVersion: 9.4.4
+            compilerVersion: 9.4.5
             setup-method: ghcup
             allow-failure: false
-          - compiler: ghc-9.2.6
+          - compiler: ghc-9.2.8
             compilerKind: ghc
-            compilerVersion: 9.2.6
+            compilerVersion: 9.2.8
             setup-method: ghcup
             allow-failure: false
           - compiler: ghc-9.0.2

--- a/indexed-profunctors/CHANGELOG.md
+++ b/indexed-profunctors/CHANGELOG.md
@@ -1,3 +1,6 @@
+# indexed-profunctors-0.1.1.1 (2023-06-22)
+* Add INLINE pragmas to small functions that really should inline
+
 # indexed-profunctors-0.1.1 (2021-04-09)
 * Remove unnecessary INLINE pragmas
 

--- a/indexed-profunctors/indexed-profunctors.cabal
+++ b/indexed-profunctors/indexed-profunctors.cabal
@@ -1,13 +1,13 @@
 cabal-version: 2.2
 name:          indexed-profunctors
-version:       0.1.1
+version:       0.1.1.1
 license:       BSD-3-Clause
 license-file:  LICENSE
 build-type:    Simple
 maintainer:    optics@well-typed.com
 author:        Adam Gundry, Andres Löh, Andrzej Rybczak, Oleg Grenrus
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Utilities for indexed profunctors
 category:      Data, Optics, Lenses, Profunctors
 description:

--- a/metametapost/metametapost.cabal
+++ b/metametapost/metametapost.cabal
@@ -5,7 +5,7 @@ license:       BSD-3-Clause
 license-file:  LICENSE
 build-type:    Simple
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 maintainer:    oleg@well-typed.com
 synopsis:      Generate optics documentation diagrams
 category:      Optics, Examples

--- a/optics-core/CHANGELOG.md
+++ b/optics-core/CHANGELOG.md
@@ -1,3 +1,6 @@
+# optics-core-0.4.1.1 (2023-06-22)
+* Add INLINE pragmas to small functions that really should inline
+
 # optics-core-0.4.1 (2022-03-22)
 * Add support for GHC-9.2
 * Add `is` ([#410](https://github.com/well-typed/optics/pull/410))

--- a/optics-core/optics-core.cabal
+++ b/optics-core/optics-core.cabal
@@ -1,13 +1,13 @@
 cabal-version: 2.2
 name:          optics-core
-version:       0.4.1
+version:       0.4.1.1
 license:       BSD-3-Clause
 license-file:  LICENSE
 build-type:    Simple
 maintainer:    optics@well-typed.com
 author:        Adam Gundry, Andres Löh, Andrzej Rybczak, Oleg Grenrus
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Optics as an abstract interface: core definitions
 category:      Data, Optics, Lenses
 description:

--- a/optics-extra/optics-extra.cabal
+++ b/optics-extra/optics-extra.cabal
@@ -7,7 +7,7 @@ build-type:    Simple
 maintainer:    optics@well-typed.com
 author:        Andrzej Rybczak
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Extra utilities and instances for optics-core
 category:      Data, Optics, Lenses
 description:

--- a/optics-sop/optics-sop.cabal
+++ b/optics-sop/optics-sop.cabal
@@ -7,7 +7,7 @@ build-type:    Simple
 maintainer:    optics@well-typed.com
 author:        Adam Gundry, Andres Löh, Andrzej Rybczak
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Optics for generics-sop, and using generics-sop
 category:      Data, Optics, Lenses, Generics
 description:

--- a/optics-th/optics-th.cabal
+++ b/optics-th/optics-th.cabal
@@ -7,7 +7,7 @@ build-type:    Simple
 maintainer:    optics@well-typed.com
 author:        Andrzej Rybczak
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Optics construction using TemplateHaskell
 category:      Data, Optics, Lenses
 description:

--- a/optics-vl/optics-vl.cabal
+++ b/optics-vl/optics-vl.cabal
@@ -7,7 +7,7 @@ build-type:    Simple
 maintainer:    optics@well-typed.com
 author:        Andrzej Rybczak
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Utilities for compatibility with van Laarhoven optics
 category:      Data, Optics, Lenses
 description:

--- a/optics/CHANGELOG.md
+++ b/optics/CHANGELOG.md
@@ -1,3 +1,6 @@
+# optics-0.4.2.1 (2023-06-22)
+* Fix compilation of tests with GHC 9.4 and 9.6
+
 # optics-0.4.2 (2022-05-19)
 * Allow `transformers-0.6` and `mtl-2.3`
 

--- a/optics/optics.cabal
+++ b/optics/optics.cabal
@@ -1,13 +1,13 @@
 cabal-version:   2.2
 name:            optics
-version:         0.4.2
+version:         0.4.2.1
 license:         BSD-3-Clause
 license-file:    LICENSE
 build-type:      Simple
 maintainer:      optics@well-typed.com
 author:          Adam Gundry, Andres Löh, Andrzej Rybczak, Oleg Grenrus
 tested-with:     GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                  || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                  || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:        Optics as an abstract interface
 category:        Data, Optics, Lenses
 description:

--- a/template-haskell-optics/template-haskell-optics.cabal
+++ b/template-haskell-optics/template-haskell-optics.cabal
@@ -7,7 +7,7 @@ maintainer:    optics@well-typed.com
 author:        Andrzej Rybczak
 cabal-version: 1.18
 tested-with:   GHC ==8.2.2 || ==8.4.4 || ==8.6.5 || ==8.8.4 || ==8.10.7
-                || ==9.0.2 || ==9.2.6 || ==9.4.4 || ==9.6.1, GHCJS ==8.4
+                || ==9.0.2 || ==9.2.8 || ==9.4.5 || ==9.6.2, GHCJS ==8.4
 synopsis:      Optics for template-haskell types
 category:      Data, Optics, Lenses
 description:

